src/main/java/core/basesyntax/HelloWorld.java

```
@@ -1,9 +1,60 @@
-    // HINT: In the `public static void main(String[] args)` it is better to create instances of your classes, 
-    // and call their methods, but do not write any business logic in the `main` method!
+    public static void main(String[] args) {
+        CsvFileReaderService csvFileReaderService = new CsvFileReaderServiceImpl();
+        CsvFileWriterService csvFileWriterService = new CsvFileWriterServiceImpl();
+        ReportGeneratorService reportGeneratorService = new ReportGeneratorServiceImpl();
+
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlers = new HashMap<>();
+        operationHandlers.put(FruitTransaction.Operation.BALANCE, new BalanceOperation());
+        operationHandlers.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation());
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation());
+        operationHandlers.put(FruitTransaction.Operation.RETURN, new ReturnOperation());
+
+        ShopService shopService = new ShopServiceImpl(operationHandlers);
+
+        try {
+            List<String> inputReport = csvFileReaderService.readFromFile(
+                    "src/main/resources/reportToRead.csv");
+            List<FruitTransaction> transactions = inputReport.stream()
+                    .skip(1)
+                    .map(line -> line.split(","))
+                    .map(parts -> new FruitTransaction(
+                            FruitTransaction.Operation.fromCode(parts[0]),
+                            parts[1],
+                            Integer.parseInt(parts[2])
+                    ))
+                    .collect(Collectors.toList());
+
+            shopService.processTransactions(transactions, Storage.getInventory());
+
+            String resultingReport = reportGeneratorService.generateReport(Storage.getInventory());
+
+            csvFileWriterService.writeToFile(resultingReport, "src/main/resources/finalReport.csv");
+        } catch (IOException e) {
```

Don't ignore exceptions in try-catch blocks. Instead of just rethrowing the exception, handle it properly or rethrow it with a more informative message about the context of the error.